lib/backend/src/Obelisk/Backend.hs

```diff
@@ -233,21 +268,35 @@ runBackendWith
   -> Backend backendRoute frontendRoute
   -> Frontend (R frontendRoute)
   -> IO ()
-runBackendWith (BackendConfig runSnap staticAssets ghcjsWidgets) backend frontend = case checkEncoder $ _backend_routeEncoder backend of
+runBackendWith (BackendConfig runSnap staticAssets ghcjsAssets ghcjsWidgets) backend frontend = case checkEncoder $ _backend_routeEncoder backend of
   Left e -> fail $ "backend error:\n" <> T.unpack e
   Right validFullEncoder -> do
     publicConfigs <- getPublicConfigs
+    enableWasm <- checkWasmFile
     _backend_run backend $ \serveRoute ->
       runSnap $
         getRouteWith validFullEncoder >>= \case
           Identity r -> case r of
             FullRoute_Backend backendRoute :/ a -> serveRoute $ backendRoute :/ a
             FullRoute_Frontend obeliskRoute :/ a ->
-              serveDefaultObeliskApp routeToUrl (($ allJsUrl) <$> ghcjsWidgets) (serveStaticAssets staticAssets) frontend publicConfigs $
+              serveObeliskApp routeToUrl widgets (serveStaticAssets staticAssets) frontendApp publicConfigs $
                 obeliskRoute :/ a
               where
                 routeToUrl (k :/ v) = renderObeliskRoute validFullEncoder $ FullRoute_Frontend (ObeliskRoute_App k) :/ v
                 allJsUrl = renderAllJsPath validFullEncoder
+                mWasmAssets = if enableWasm
+                  then Just (renderWasmPathRoot validFullEncoder, defaultWasmAssets)
+                  else Nothing
+                widgets = ($ GhcjsWasmAssets allJsUrl mWasmAssets) <$> ghcjsWidgets
+                frontendApp = GhcjsApp
+                  { _ghcjsApp_compiled = ghcjsAssets
+                  , _ghcjsApp_value = frontend
+                  }
+  where
+    checkWasmFile = do
+      let base = _staticAssets_processed ghcjsAssets
+          p = T.unpack $ _wasmAssets_wasmFile defaultWasmAssets
+      isJust <$> getAssetPath base p
```

Comment:
nit: Binding to name `p` doesn't communicate much so might as well inline that.

Reply:
Agreed